Include a short section about integrating discord (#123)
spacelift-io/user-documentation: docs/concepts/policy/notification-policy.md

```

[View the example in the rego playground](https://play.openpolicyagent.org/p/fbfiiYEots){: rel="nofollow"}.

> Using custom webhook requests also makes it quite easy to integrate spacelift with any third party webhook consumers.

Reviewer: ```suggestion
Using custom webhook requests also makes it quite easy to integrate Spacelift with any third-party webhook consumer.
```

#### Custom webhook requests in action

##### Discord integration

> Discord can be integrated to receive updates about Spacelift by simply creating a new webhook endpoint in your discord server's integrations

Reviewer: ```suggestion
Discord can be integrated to receive updates about Spacelift by simply creating a new webhook endpoint in your Discord server's integrations
```

section and providing that as the endpoint when creating a new [named webhook](../../integrations/webhooks.md).

!!! info
    <!-- markdown-link-check-disable -->

>     For more information about making discord webhooks follow their [official webhook guide](https://support.discord.com/hc/en-us/articles/228383668).

Author: This link works but linter is now allowed to access it.

Reviewer: @tomasmik Are you sure? I ran the linter locally, after removing the comment, and it was able to test the link.

Author: Not sure how the linter works, it could be that it only checks new/changed lines that will be committed and ignores any old ones. That would make it so that it still works for you.

Here's what I got:
<img width="640" alt="image" src="https://user-images.githubusercontent.com/40318863/212358844-4c74275f-c57d-4514-9d20-1743c414000a.png">

Reviewer: Yes, unless you add the `--all-files` flag, the linter will only check the staged files.

I know that some servers confuse the linter for a bot and block it. It might be one of these. Ignoring that link is fine then.

>     For more information about making discord webhooks follow their [official webhook guide](https://support.discord.com/hc/en-us/articles/228383668).

Reviewer: ```suggestion
    For more information about making Discord webhooks follow the [official webhook guide](https://support.discord.com/hc/en-us/articles/228383668){: rel="nofollow"}.
```

>     For more information about making Discord webhooks follow their [official webhook guide](https://support.discord.com/hc/en-us/articles/228383668){: rel="nofollow"}..

Reviewer: One last nit-picking. 😄 

```suggestion
    For more information about making Discord webhooks follow their [official webhook guide](https://support.discord.com/hc/en-us/articles/228383668){: rel="nofollow"}.
```

After creating the webhook on both Discord and Spacelift you can define a new webhook rule like this:

```opa
# Send updates about tracked runs to discord.
webhook[wbdata] {
  endpoint := input.webhook_endpoints[_]
  endpoint.id == "YOUR_WEBHOOK_ID_HERE"
  stack := input.run_updated.stack
  run := input.run_updated.run
  wbdata := {
    "endpoint_id": endpoint.id,
    "payload": {
      "embeds": [{
        "title": "Tracked run triggered!",
        "description": sprintf("Stack: [%s](http://example.app.spacelift.tf/stack/%s)\nRun ID: [%s](http://example.app.spacelift.tf/stack/%s/run/%s)\nRun state: %s", [stack.name,stack.id,run.id,stack.id, run.id,run.state]),
        }]
     }
  }
  input.run_updated.run.type == "TRACKED"
}
```

> And that's it! You should now be receiving updates about tracked runs to your discord server:

Reviewer: ```suggestion
And that's it! You should now be receiving updates about tracked runs to your Discord server:
```
